docs/fundamentals/documentarray/embedding.md

```diff
@@ -1,130 +1,157 @@
+import tensorflow as tf
+from docarray import DocumentArray
 
-````{tab} ONNX
-
-Preliminary: you need to first export a DNN model to ONNX via API/CLI. 
-For example let's use the PyTorch one:
+docs = DocumentArray.empty(10)
+docs.tensors = np.random.rand(10, 224, 224, 3).astype('float32')
 
-```python
-data = torch.rand(1, 128)
-
-torch.onnx.export(model, data, 'mlp.onnx', 
-    do_constant_folding=True,  # whether to execute constant folding for optimization
-    input_names=['input'],  # the model's input names
-    output_names=['output'],  # the model's output names
-    dynamic_axes={
-        'input': {0: 'batch_size'},  # variable length axes
-        'output': {0: 'batch_size'},
-    })
+embed_model = tf.keras.applications.resnet50.ResNet50()
+docs.embed(embed_model=embed_model, device='cuda', device_id=0, batch_size=5)
 ```
-
-Then load it as `InferenceSession`:
- 
+````
+````{tab} ONNX ResNet50
 ```python
-import onnxruntime
+import numpy as np
+import onnxruntime as ort
+from docarray import DocumentArray
 
-model = onnxruntime.InferenceSession('mlp.onnx')
+docs = DocumentArray.empty(10)
+docs.tensors = np.random.rand(10, 3, 224, 224).astype('float32')
+# please download model here: https://github.com/onnx/models/blob/main/vision/classification/resnet/model/resnet50-v1-12.onnx
+ort_session = ort.InferenceSession('~your-path/resnet50-v1-12.onnx')
+docs.embed(embed_model=ort_session, device='cuda', device_id=0, batch_size=5)
 ```
 ````
-
-````{tab} Paddle
-
+````{tab} PaddlePaddle ResNet50
 ```python
+import numpy as np
 import paddle
+from docarray import DocumentArray
 
-model = paddle.nn.Sequential(
-    paddle.nn.Linear(
-        in_features=128,
-        out_features=128,
-    ),
-    paddle.nn.ReLU(),
-    paddle.nn.Linear(in_features=128, out_features=32),
-)
+docs = DocumentArray.empty(10)
+docs.tensors = np.random.rand(10, 3, 224, 224).astype('float32')
+
+embed_model = paddle.vision.models.resnet50()
+docs.embed(embed_model=embed_model, device='cuda', device_id=0, batch_size=5)
 ```
 ````
-
-Now, you can create the embeddings:
-
+````{tab} Timm ResNet50
 ```python
-docs.embed(model)
+import numpy as np
+import timm
+from docarray import DocumentArray
+
+docs = DocumentArray.empty(10)
+docs.tensors = np.random.rand(10, 3, 224, 224).astype('float32')
 
-print(docs.embeddings)
+embed_model = timm.create_model('resnet50')
+docs.embed(embed_model=embed_model, device='cuda', device_id=0, batch_size=5)
 ```
+````
 
-```text
-tensor([[-0.1234,  0.0506, -0.0015,  0.1154, -0.1630, -0.2376,  0.0576, -0.4109,
-          0.0052,  0.0027,  0.0800, -0.0928,  0.1326, -0.2256,  0.1649, -0.0435,
-         -0.2312, -0.0068, -0.0991,  0.0767, -0.0501, -0.1393,  0.0965, -0.2062,
+```{important}
+In practice, when using a pre-trained network such as ResNet,
+please remove the last fully-connected layer.
+This is because the last layers are used to optimize objectives such as class scores,
+not suitable for serving as embedding layers.
```

Comment:
```suggestion
In practice, when using a pre-trained network such as ResNet,
you should remove the last fully-connected layer.
This is because the last layers are used to optimize objectives such as class scores,
not to serve as embedding layers.
```